src/tests/ftest/telemetry/telemetry_pool_metrics.py

```diff
@@ -0,0 +1,180 @@
+        self.log.debug("Upper Bound: %s", upper_bound)
+        # Check if the actual value is within range
+        if lower_bound <= actual_value <= upper_bound:
+            return True
+
+        return False
+
+    def verify_pool_metrics(self, pool_metrics_list, metrics_data,
+                            expected_values):
+        """ Verify telemetry pool metrics from metrics_data.
+
+        Args:
+            pool_metrics_list (list): list of telemetry pool metrics.
+            metrics_data (dict): a dictionary of host keys linked to a
+                                 list of pool metric names. Contains
+                                 data before and after read/write.
+            expected_values (list): List of expected values for different
+                                    pool metrics.
+
+        """
+        # initializing method params
+        total = []
+        final_value = []
+        # collecting data to be verified
+        for name in sorted(pool_metrics_list):
+            self.log.debug("  --telemetry metric: %s", name)
+            for key in sorted(metrics_data):
+                m_data = metrics_data[key]
+                total.append(0)
+                total[key] = 0
+                for host in sorted(m_data[name]):
+                    for rank in sorted(m_data[name][host]):
+                        for target in sorted(m_data[name][host][rank]):
+                            value = m_data[name][host][rank][target]
+                            # collecting total value for all targets
+                            total[key] = total[key] + value
+            # subtracting the total value after read/write from total value
+            # before read/write
+            final_value.append(total[1] - total[0])
+            # performing verification now
+            if "xferred" in name:
+                # check if oclass has any replication and get the replication
+                # numeric value, else use 1.
+                if True in [char.isdigit() for char in
+                            self.ior_cmd.dfs_oclass.value]:
+                    replication = re.search(
+                        r'\d+', self.ior_cmd.dfs_oclass.value).group()
+                else:
+                    replication = 1
+                # multiply the written amount by replication number
+                # to get total expected written amount for only
+                # write operation.
+                if "update" in name:
+                    total_amount_written = (self.ior_cmd.block_size.value *
+                                            int(replication))
+                else:
+                    total_amount_written = self.ior_cmd.block_size.value
+                # get difference between actual written value and expected
+                # written value
+                final_value[-1] = final_value[-1] - total_amount_written
+                # perform check
+                if final_value[-1] > expected_values[1]:
```

Comment:
So if I'm understanding this correctly, we're checking that the diff is not greater than some margin of error (the DFS chunk size)? might be worth some name changes or comments indicating what we expect to find in expected_values[1] vs. expected_values[0]

Reply:
The changes are clearer, thanks!